### PR TITLE
Adapt to the new kernel HSM driver

### PR DESCRIPTION
**Pull request by shuox**
A new HSM kernel driver has been merged in Linux 5.12 already. To adapt
the new driver, both devicemodel and hypervisor need to use the acrn.h
interface header file which is exported from the new driver.

This patch introduce the acrn.h and remove the old vhm_ioctl_defs.h.
Meanwhile, lots of structures, functions, variables are renamed. They
includes,

struct vhm_request		->	struct acrn_io_request
union vhm_request_buffer	->	struct acrn_io_request_buffer
struct pio_request		->	struct acrn_pio_request
struct mmio_request		->	struct acrn_mmio_request
struct acrn_set_vcpu_regs	->	struct acrn_vcpu_regs
struct acrn_vcpu_regs		->	struct acrn_regs
struct acrn_create_vm		->	struct acrn_vm_creation
struct ioreq_notify		->	struct acrn_ioreq_notify
struct vm_memmap		->	struct acrn_vm_memmap
struct acrn_assign_pcidev	->	struct acrn_pcidev
struct acrn_emul_dev		->	struct acrn_vdev
struct ic_ptdev_irq		->	struct acrn_ptdev_irq
struct acrn_vm_config		->	struct acrn_vm_config_header
(DM only)
struct platform_info		->	struct acrn_platform_info
struct cpu_px_data		->	struct acrn_pstate_data
struct cpu_cx_data		->	struct acrn_cstate_data
enum pm_cmd_type		->	enum acrn_pm_cmd_type
struct acpi_generic_address	->	struct acrn_acpi_generic_address
struct hc_platform_info		->	struct acrn_platform_info

VHM_REQ_PIO_INVAL		->	IOREQ_PIO_INVAL
VHM_REQ_MMIO_INVAL		->	IOREQ_MMIO_INVAL
REQ_PORTIO			->	ACRN_IOREQ_TYPE_PORTIO
REQ_MMIO			->	ACRN_IOREQ_TYPE_MMIO
REQ_PCICFG			->	ACRN_IOREQ_TYPE_PCICFG
REQ_WP				->	ACRN_IOREQ_TYPE_WP

REQUEST_READ			->	ACRN_IOREQ_DIR_READ
REQUEST_WRITE			->	ACRN_IOREQ_DIR_WRITE
REQ_STATE_PROCESSING		->	ACRN_IOREQ_STATE_PROCESSING
REQ_STATE_PENDING		->	ACRN_IOREQ_STATE_PENDING
REQ_STATE_COMPLETE		->	ACRN_IOREQ_STATE_COMPLETE
REQ_STATE_FREE			->	ACRN_IOREQ_STATE_FREE

IC_CREATE_VM			->	ACRN_IOCTL_CREATE_VM
IC_CREATE_IOREQ_CLIENT		->	ACRN_IOCTL_CREATE_IOREQ_CLIENT
IC_DESTROY_IOREQ_CLIENT		->	ACRN_IOCTL_DESTROY_IOREQ_CLIENT
IC_ATTACH_IOREQ_CLIENT		->	ACRN_IOCTL_ATTACH_IOREQ_CLIENT
IC_NOTIFY_REQUEST_FINISH	->	ACRN_IOCTL_NOTIFY_REQUEST_FINISH
IC_DESTROY_VM			->	ACRN_IOCTL_DESTROY_VM
IC_SET_MEMSEG			->	ACRN_IOCTL_SET_MEMSEG
IC_UNSET_MEMSEG			->	ACRN_IOCTL_UNSET_MEMSEG
IC_START_VM			->	ACRN_IOCTL_START_VM
IC_PAUSE_VM			->	ACRN_IOCTL_PAUSE_VM
IC_RESET_VM			->	ACRN_IOCTL_RESET_VM
IC_CLEAR_VM_IOREQ		->	ACRN_IOCTL_CLEAR_VM_IOREQ
IC_INJECT_MSI			->	ACRN_IOCTL_INJECT_MSI
IC_SET_IRQLINE			->	ACRN_IOCTL_SET_IRQLINE
IC_ASSIGN_PCIDEV		->	ACRN_IOCTL_ASSIGN_PCIDEV
IC_DEASSIGN_PCIDEV		->	ACRN_IOCTL_DEASSIGN_PCIDEV
IC_ASSIGN_MMIODEV		->	ACRN_IOCTL_ASSIGN_MMIODEV
IC_DEASSIGN_MMIODEV		->	ACRN_IOCTL_DEASSIGN_MMIODEV
IC_ADD_HV_VDEV			->	ACRN_IOCTL_CREATE_VDEV
IC_REMOVE_HV_VDEV		->	ACRN_IOCTL_DESTROY_VDEV
IC_SET_PTDEV_INTR_INFO		->	ACRN_IOCTL_SET_PTDEV_INTR
IC_RESET_PTDEV_INTR_INFO	->	ACRN_IOCTL_RESET_PTDEV_INTR
IC_SET_VCPU_REGS		->	ACRN_IOCTL_SET_VCPU_REGS
IC_PM_GET_CPU_STATE		->	ACRN_IOCTL_PM_GET_CPU_STATE
IC_VM_INTR_MONITOR		->	ACRN_IOCTL_VM_INTR_MONITOR
IC_EVENT_IOEVENTFD		->	ACRN_IOCTL_IOEVENTFD
IC_EVENT_IRQFD			->	ACRN_IOCTL_IRQFD
IC_GET_PLATFORM_INFO		->	ACRN_IOCTL_GET_PLATFORM_INFO

HYPERVISOR_CALLBACK_VHM_VECTOR	->	HYPERVISOR_CALLBACK_HSM_VECTOR
MAX_PLATFORM_LAPIC_IDS		->	ACRN_PLATFORM_LAPIC_IDS_MAX
VM_MEMMAP_SYSMEM		->	ACRN_MEMMAP_RAM
VM_MMIO				->	ACRN_MEMMAP_MMIO
PROT_ALL			->	ACRN_MEM_ACCESS_RWX
gpa				->	user_vm_pa
hpa				->	service_vm_pa
IRQ_INTX			->	ACRN_PTDEV_IRQ_INTX
IRQ_MSI				->	ACRN_PTDEV_IRQ_MSI
QUIRK_PTDEV			->	ACRN_PTDEV_QUIRK_ASSIGN

PMCMD_GET_PX_CNT		->	ACRN_PMCMD_GET_PX_CNT
PMCMD_GET_CX_CNT		->	ACRN_PMCMD_GET_CX_CNT
PMCMD_GET_PX_DATA		->	ACRN_PMCMD_GET_PX_DATA
PMCMD_GET_CX_DATA		->	ACRN_PMCMD_GET_CX_DATA

arch_fire_vhm_interrupt()	->	arch_fire_hsm_interrupt()
get_vhm_notification_vector()	->	get_hsm_notification_vector()
set_vhm_notification_vector()	->	set_hsm_notification_vector()
acrn_vhm_notification_vector	->	acrn_hsm_notification_vector
get_vhm_req_state()		->	get_io_req_state()
set_vhm_req_state()		->	set_io_req_state()

Below structures have slight difference with former ones.

  struct acrn_ioreq_notify
  struct acrn_vm_memmap
  struct acrn_platform_info
  struct acrn_vdev
  strcut io_request

Please note!!

This patch will break the regression test by 'git bisect'.  Because of
the change of the driver interfaces, the former DM&HV need work with old
VHM kernel driver while the newer DM&HV (have this patch) need work with
new HSM driver.

### CODE REVIEW

**Comment by dongyaozu** [comment]
@ywan170 can u review the DM part? I am fine with the patch

**Review by ywan170 — APPROVED** [review]

**Comment by wenlingz** [comment]
@dbkinder We need to take time to  modify doc part, because we switched kernel to 5.10 from 5.4, there are some dependency, currently make doc fail. So we will  merge this PR firstly, next shuo will provide new PR for doc part, thanks for your understanding.

**Comment by dbkinder** [comment]
@wenlingz @NanlinXie if you merge this while it has a doc build error, all further CI runs will fail until the doc build error is fixed :(

File: doc/.known-issues/doc/kerneldoc.conf [hunk 1]
```diff
@@ -6,3 +6,5 @@
 ^Invalid C declaration:.*
 ^  long virtio_.*
 ^[- \t]*\^
+^(?P<filename>[-._/\w]+/acrn-kernel/drivers/gpu/drm/i915/gvt/mpt.h):(?P<lineno>[0-9]+): warning: Function parameter .* not described in .*
```
Comment (dbkinder) [inline doc/.known-issues/doc/kerneldoc.conf:9]: The right fix is to fix the incorrect code kerneldoc comments, not to hide the error message.  Remove these two additional lines (they're incorrectly done anyway).

**Review by dbkinder — CHANGES_REQUESTED** [review]
The kerneldoc comments in ``acrn-kernel/drivers/gpu/drm/i915/gvt/mpt.h`` need to be fixed instead of ignoring the error!

**Review by dbkinder — CHANGES_REQUESTED** [review]
Can you also look as these un-edited references to VHM in:

- developer-guides/GVT-g-porting.rst
- developer-guides/hld/hld-APL_GVT-g.rst
- api/GVT-g_api.rst

**Comment by wenlingz** [comment]
try verify EHL

**Comment by wenlingz** [comment]
@shuox please refer build fail:
linux/acrn-devicemodel/6283-1626090355/build/devicemodel/log/disk_logger.o
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/acpi.h:32,
|                  from hw/pci/virtio/virtio_gpio.c:21:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'
|   237 |   __u16 cpu_num;
|       |         ^~~~~~~
| Makefile:247: recipe for target '/home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/build/devicemodel/hw/platform/atkbdc.o' failed
| make[1]: *** [/home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/build/devicemodel/hw/platform/atkbdc.o] Error 1
| make[1]: *** Waiting for unfinished jobs....
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/vmmapi.h:37,
|                  from core/monitor.c:25:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'
|   237 |   __u16 cpu_num;
|       |         ^~~~~~~
| Makefile:247: recipe for target '/home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/build/devicemodel/core/monitor.o' failed
| make[1]: *** [/home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/build/devicemodel/core/monitor.o] Error 1
| Makefile:247: recipe for target '/home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/build/devicemodel/hw/pci/irq.o' failed
| make[1]: *** [/home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/build/devicemodel/hw/pci/irq.o] Error 1
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/vmmapi.h:37,
|                  from hw/platform/acpi/rtct.c:20:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'
|   237 |   __u16 cpu_num;
|       |         ^~~~~~~
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/vmmapi.h:37,
|                  from hw/pci/lpc.c:40:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'
|   237 |   __u16 cpu_num;
|       |         ^~~~~~~
| Makefile:247: recipe for target '/home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/build/devicemodel/hw/platform/acpi/rtct.o' failed
| make[1]: *** [/home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/build/devicemodel/hw/platform/acpi/rtct.o] Error 1
| Makefile:247: recipe for target '/home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/build/devicemodel/hw/pci/lpc.o' failed
| make[1]: *** [/home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/build/devicemodel/hw/pci/lpc.o] Error 1
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/vmmapi.h:37,
|                  from hw/pci/core.c:40:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'
|   237 |   __u16 cpu_num;
|       |         ^~~~~~~
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/vmmapi.h:37,
|                  from hw/mmio/core.c:18:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'
|   237 |   __u16 cpu_num;
|       |         ^~~~~~~
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/vmmapi.h:37,
|                  from hw/platform/rtc.c:36:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'
|   237 |   __u16 cpu_num;
|       |         ^~~~~~~
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/vmmapi.h:37,
|                  from hw/pci/virtio/virtio_mei.c:36:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'
|   237 |   __u16 cpu_num;
|       |         ^~~~~~~
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/vmmapi.h:37,
|                  from core/sw_load_vsbl.c:34:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'
|   237 |   __u16 cpu_num;
|       |         ^~~~~~~
| Makefile:247: recipe for target '/home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/build/devicemodel/hw/mmio/core.o' failed
| make[1]: *** [/home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/build/devicemodel/hw/mmio/core.o] Error 1
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/vmmapi.h:37,
|                  from hw/pci/xhci.c:96:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'
|   237 |   __u16 cpu_num;
|       |         ^~~~~~~
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/vmmapi.h:37,
|                  from hw/platform/hpet.c:42:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'
|   237 |   __u16 cpu_num;
|       |         ^~~~~~~
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/vmmapi.h:37,
|                  from hw/platform/tpm/tpm_crb.c:15:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'
|   237 |   __u16 cpu_num;
|       |         ^~~~~~~
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/vmmapi.h:37,
|                  from hw/platform/debugexit.c:33:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'
|   237 |   __u16 cpu_num;
|       |         ^~~~~~~
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/vmmapi.h:37,
|                  from hw/pci/ptm.c:20:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'
|   237 |   __u16 cpu_num;
|       |         ^~~~~~~
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/vmmapi.h:37,
|                  from hw/pci/wdt_i6300esb.c:18:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'
|   237 |   __u16 cpu_num;
|       |         ^~~~~~~
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/vmmapi.h:37,
|                  from hw/platform/pit.c:38:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'
|   237 |   __u16 cpu_num;
|       |         ^~~~~~~
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/vmmapi.h:37,
|                  from hw/platform/acpi/acpi_pm.c:12:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'
|   237 |   __u16 cpu_num;
|       |         ^~~~~~~
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/vmmapi.h:37,
|                  from hw/pci/npk.c:95:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'
|   237 |   __u16 cpu_num;
|       |         ^~~~~~~
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/vmmapi.h:37,
|                  from hw/platform/cmos_io.c:17:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'
|   237 |   __u16 cpu_num;
|       |         ^~~~~~~
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/vmmapi.h:37,
|                  from core/sw_load_bzimage.c:34:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'
|   237 |   __u16 cpu_num;
|       |         ^~~~~~~
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/vmmapi.h:37,
|                  from hw/pci/uart.c:39:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'
|   237 |   __u16 cpu_num;
|       |         ^~~~~~~
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/vmmapi.h:37,
|                  from core/sw_load_ovmf.c:37:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'
|   237 |   __u16 cpu_num;
|       |         ^~~~~~~
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/vmmapi.h:37,
|                  from hw/platform/tpm/tpm.c:13:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'
|   237 |   __u16 cpu_num;
|       |         ^~~~~~~
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/vmmapi.h:37,
|                  from core/pm.c:9:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'
|   237 |   __u16 cpu_num;
|       |         ^~~~~~~
| In file included from /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/vmmapi.h:37,
|                  from core/sw_load_elf.c:44:
| /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/6283-1626090355/git/devicemodel/include/public/hsm_ioctl_defs.h:237:9: error: duplicate member 'cpu_num'

File: doc/.known-issues/doc/kerneldoc.conf (hunk 1 above)
Comment (shuox) [reply doc/.known-issues/doc/kerneldoc.conf:9]: Hi @dbkinder,
It's in upstream i915 driver. Previously, we keep an offline patch in acrn-kernel(5.4 branch) to pass this doc build. Now, after switching to 5.10 base, PKT kernel doesn't accept non-upstream patch anymore. 
Of course, we can upstream this doc fix in i915 driver. However, it's another repo (kernel) business. While our acrn-hypervisor will be blocked until the patch is accepted by Linux community and backported.

Such dependency is not good. This change can remove the dependency.

**Comment by ywan170** [comment]
> > Can you also look as these un-edited references to VHM in:
> > 
> > * developer-guides/GVT-g-porting.rst
> > * developer-guides/hld/hld-APL_GVT-g.rst
> > * api/GVT-g_api.rst
> 
> HSM doesn't support GVT-g at this moment. I don't know if we will continue to support GVT-g in future as we have SRIOV now.
> If we will support,
> 
> 1. We need upstream acrngt code to Linux community.
> 2. update these documents.
>    If we won't,
> 3. Go through the documents and mark them as obsolete.
> 
> Before the decision, I just keep them and fix the 'doc' build error.

The GVT-G will be dropped from TGL. I don't think we will upstream the acrngt..

**Comment by shuox** [comment]
> > > Can you also look as these un-edited references to VHM in:
> > > 
> > > * developer-guides/GVT-g-porting.rst
> > > * developer-guides/hld/hld-APL_GVT-g.rst
> > > * api/GVT-g_api.rst
> > 
> > 
> > HSM doesn't support GVT-g at this moment. I don't know if we will continue to support GVT-g in future as we have SRIOV now.
> > If we will support,
> > 
> > 1. We need upstream acrngt code to Linux community.
> > 2. update these documents.
> >    If we won't,
> > 3. Go through the documents and mark them as obsolete.
> > 
> > Before the decision, I just keep them and fix the 'doc' build error.
> 
> The GVT-G will be dropped from TGL. I don't think we will upstream the acrngt..

@ywan170 @dbkinder Do you think it's better to remove all GVT-g related documents? Or just mark them as obsolete?

File: doc/.known-issues/doc/kerneldoc.conf (hunk 1 above)
Comment (dbkinder) [reply doc/.known-issues/doc/kerneldoc.conf:9]: Can we make the comment changes in the mpt.h file in the acrn-kernel repo we maintain? This is what was done in the past for our API document generation.  The change to the known-issues pattern hides the error message, but the API documentation won't be generated correctly.

Comment (shuox) [reply doc/.known-issues/doc/kerneldoc.conf:9]: I can do that if the doc build using acrn-kernel.

**Comment by wenlingz** [comment]
OK to verify doc

**Comment by gvancuts** [comment]
> > The GVT-G will be dropped from TGL. I don't think we will upstream the acrngt..
> 
> @ywan170 @dbkinder Do you think it's better to remove all GVT-g related documents? Or just mark them as obsolete?

I think we should remove these altogether, GVT-g is not supported in ACRN anymore so if users want to use it, they'll have to use a previous version (and the GVT-g doc will be included in these).

**Comment by NanlinXie** [comment]
I agree to remove the GVT-g doc since it is out of date for now.  And we also need to review current document build dependancy for kernel source file, it is not good experience to require user to clone acrn-kernel source code to build acrn doc. @dbkinder , is it possible we link to upstream kernel source file instead of local source file to build doc?

**Comment by ywan170** [comment]
Agree. Let's remove it. Thanks @gvancuts

**Comment by dbkinder** [comment]
If we remove all GVT-g documentation from ACRN, that would completely remove any dependency on requiring a local copy of the  kernel (acrn-kernel) for documentation generation.  Specifically we'd remove ``api/GVT-g_api.rst``, remove sections (noted in this PR) that are now obsolete, and tweak any references to that documentation (there are only a couple). @shuox can remove the sections instead of adding "obsolete" to the section titles in this PR.   I'll submit a separate PR to remove the API documentation and references to it.

**Comment by dbkinder** [comment]
@shuox PR #6294 removed the GVT-d documentation that your PR updated.  Please take a look at that PR and resolve the conflicts by just removing your edits to those files.

**Comment by shuox** [comment]
> @shuox PR #6294 removed the GVT-d documentation that your PR updated. Please take a look at that PR and resolve the conflicts by just removing your edits to those files.

Thanks, rebase done.

**Comment by wenlingz** [comment]
OK to verify doc

**Pull request merged**